average.js

```diff
@@ -1,6 +1,14 @@
-
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  let totalNotNulls = numbers.length;
+  return (
+    numbers.reduce((p, c) => {
+      if (c) {
+        return p + c;
+      }
+      totalNotNulls--;
+      return p;
+    }, 0) / totalNotNulls
+  );
 }
 
-module.exports = {average};
+module.exports = { average };
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
The functionality accepts a sequence of readings from an IoT sensor. The sensor sends `NaN` when something goes wrong in measurement, which we have ignored.
Problem: If the sensor sends a lot of NaNs and very few readings, something is wrong. However, the `average` function ignores all the NaNs and still computes the average, as if nothing went wrong.
We need to add functionality to the average function to return `NaN`, when most of the input-numbers are `NaN`.
You need to:

- Reason out what 'most of the numbers' means: E.g., determine the number of NaNs in the input which would indicate that the sensor is unreliable. Mention any references you used to arrive at this logic
- Add a test that supplies such inputs and expects the output to be NaN
- Add code that satisfies this functionality, while not breaking existing functionality.